bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java

```diff
@@ -531,6 +556,10 @@ private byte[] buildRequestArray(SocketAddress address, String request) throws j
                         requestFrame.write((byte) ((dPort >> 8) & 0xff));
                         requestFrame.write((byte) (dPort & 0xff));
                         break;
+                    case "$" + PARAMETER_SRC_MAC:
+                        byte[] mac = macBytesFrom(sock);
```

Comment:
The problem is that `String.format` always creates a character representation of the given byte i.e. 0x01 (byte) is converted to 0x31 (character).

One idea would be that if there is no `%` character in the format specifier then no byte=>character substitution would occur; so an empty format string would result in a no-op conversion of the mac bytes. => WDYT?

I will make a commit showing the above idea so that you can explore it.

Reply:
Another idea could be to replace the format string for both cases just with a delimiter.
Less flexible, but maybe also sufficient.

Reply:
> replace the format string for both cases just with a delimiter.

That is more or less what I meant.

```
String format;
byte [] test = new byte[] {1, 2, 3, 4, 5, 6, 7, -1};
format = ""; => array of bytes as above
format = ":"; => array of bytes as above interspersed with `:` delimiter
format = "%02x"; => "01020304050607ff"
format = "%02x-"; => "01-02-03-04-05-06-07-ff"
format = "%02X:"; => "01:02:03:04:05:06:07:FF"
```

Reply:
When I proposed using a delimiter above, I thought it could be used instead of format strings.
But it seems the flexibility of format strings is helpful, e.g. to format the hex numbers in uppercase or lowercase.

However, formatted MAC addresses seem to be a realistic use case only for text requests. So converting the single characters of a formatted MAC string is out of scope. If we allow only binary data in the binary requests, this is in line with the approach we use for the IP addresses etc. as well - inserting the raw bytes into the stream.

I am fine keeping the code as it is, as long as we cover it in the PR in the docs repo.

Reply:
@holgerfriedrich .. you were a bit fast with your merge .. the format string depends on 1) zero or one delimiter characters (there is no check for two or more), and 2) delimiters can only be in the `Set.of(':', ',', '-')` .. I was working on a tweak for these (minor) issues .. which I will submit as a new PR.

Reply:
See #4481